tests/test_data_entry_flow.py

```diff
@@ -473,6 +474,10 @@ def capture_events(event: Event) -> None:
         events.append(event)
         event_received_evt.set()
 
+    @callback
+    def capture_progress_update(event: Event) -> None:
```

Comment:
We can use this helper:
https://github.com/home-assistant/core/blob/eb19c7af32e34a4876aaf9bedcbcbe9cffe5299c/tests/common.py#L1590-L1602